src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -675,7 +686,12 @@ def __init__(
 @rule
 async def setup_pex_process(request: PexProcess, pex_environment: SandboxPexEnvironment) -> Process:
     pex = request.pex
-    argv = pex_environment.create_argv(pex.name, *request.argv, python=pex.python)
+    pex_bin = (
+        os.path.relpath(pex.name, request.working_directory)
+        if request.working_directory
+        else pex.name
+    )
+    argv = pex_environment.create_argv(pex_bin, *request.argv, python=pex.python)
     env = {
         **pex_environment.environment_dict(python_configured=pex.python is not None),
```

Comment:
I think `environment_dict` aught to be taking a working_directory argument to adjust PEX_ROOT directly instead of doing the fixup dance you did over in #12374.

Reply:
I'm not sure this is any better, but see what you think: #12479